File: Iniciante/C/1187 - Área Superior.c
#include <stdio.h>
int main()
{
double M[12][12],soma,media;
int i,x,y,j;
char a;
scanf("%c",&a);
soma=0;
x=0;
y=11;
    for (i=0;i<12;i++)
    {
        for (j=0;j<12;j++)
        {
        scanf("%lf",&M[i][j]);
        if (x < j && j < y)
        {
             soma = soma + M[i][j];
        }
        }
        x++;
        y--;
    }
    if (a == 'S')
    {
            printf("%.1lf\n",soma);

    }
    else if (a == 'M')
    {
            media = soma/30.0;
            printf("%.1lf\n",media);

    }

    else
    return 0;
}

File: URI/Iniciante/Python/1187 - Área Superior.py
operacao = input()
matriz = []
soma = 0
media = 0
total = 0
min = -1
max = 12

for i in range(12):
    linha = []
    for j in range(12):
        linha.append(float(input()))
    matriz.append(linha)

if operacao == "S":
    for x in range(12):
        max -= 1
        min += 1
        for y in range(12):
           if (y > min) and (y < max):
               soma += float(matriz[x][y])
    print("%.1f"%soma)
else:
    for x in range(12):
        max -= 1
        min += 1
        for y in range(12):
            if (y > min) and (y < max):
                media += float(matriz[x][y])
                total += 1
    media = media/total
    print("%.1f"%media)
